src/shared/fdset.c

```diff
@@ -133,18 +136,28 @@ bool fdset_contains(FDSet *s, int fd) {
                 return false;
         }
 
-        return !!set_get(MAKE_SET(s), FD_TO_PTR(fd));
+        return hashmap_contains(MAKE_HASHMAP(s), FD_TO_PTR(fd));
 }
 
 int fdset_remove(FDSet *s, int fd) {
```

Comment:
i think we should have separate fdset_remove() which only removes by fd, and fdset_remove_by_index() which removes by index. i.e. the fact that we do this storage should be abstracted away, and not visible to the outside